sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/csv/CSVOptions.scala

```diff
@@ -135,6 +135,8 @@ class CSVOptions(
   val positiveInf = parameters.getOrElse("positiveInf", "Inf")
   val negativeInf = parameters.getOrElse("negativeInf", "-Inf")
 
+  // Set bom to true to fix some characters are garbled when opening with Excel.
+  val bom = getBool("bom")
```

Comment:
I think you should add an assert here that encoding is always UTF-X such as UTF-8, UTF-16, UTF-16EL or UTF-16BL. I would also name it as `writeBOM`.
You will have to document this as an option at `DataFrameWrtier`, `DataStreamWriter`, `readwrite.py` and `streaming.py`.

Reply:
It seem writeBOM only support UTF-8.